src/main/java/Duke.java

```diff
@@ -1,10 +1,99 @@
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    public static String horizontalLine = "_".repeat(60) + "\n";
+    public static void main(String[] args) throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String task = "";
+        System.out.println("Hello! I'm Bot\n" + "What can I do for you?\n" + horizontalLine);
+        while (true) {
+            try {
+                task = sc.nextLine();
+                if (task.equals("bye")) {
+                    break;
+                }
+                if (task.equals("list")) {
+                    Task taskInstance = new Task("");
+                    taskInstance.printList();
+                    continue;
+                }
+                String[] elements = task.split((" "));
+                if (elements.length >= 2) {
+                    String instruction = elements[0];
+                    if (instruction.equals("mark") || instruction.equals("unmark")) {
+                        try {
+                            int no = Integer.parseInt(elements[1]);
+                            Task taskInstance = new Task("");
+                            if (instruction.equals("mark")) {
+                                taskInstance.mark(no);
+                            } else {
+                                taskInstance.unmark(no);
+                            }
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
+                            continue;
+                        }
+                    }
+                    if (instruction.equals("delete")) {
+                        try {
+                            int no = Integer.parseInt(elements[1]);
+                            Task taskInstance = new Task("");
+                            taskInstance.delete(no);
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
```

Comment:
Was wondering if this line is longer than 120 words (as per the limit in our java coding standard)?